### DISCUSSION TITLE
Respect property enumerability while snapshotting

### DISCUSSION THREAD

**Discussion by alexander-entin** [pmndrs/valtio#726] [Ideas]
Currently:

````javascript
let x = proxy(Object.defineProperty({ a: 1 }, 'b', { value: 2 }))
console.log(Object.keys(x), Object.keys(snapshot(x)))
// expected ['a'] ['a']
-> ['a'] ['a', 'b']

````
Thank you!

**Comment by dai-shi** [comment]
Nice catch. If it's reasonably feasible, it should be fixed. Would you like to give a try?

Maybe just changing this?
https://github.com/pmndrs/valtio/blob/f8a1bdcf03e2c6ec0482f6aef051400f7881c826/src/vanilla.ts#L127
